src/libraries/System.Private.Xml/src/System/Xml/Dom/XmlProcessingInstruction.cs

```diff
@@ -14,94 +14,93 @@ public class XmlProcessingInstruction : XmlLinkedNode
         private readonly string _target;
         private string _data;
 
-        protected internal XmlProcessingInstruction(string target, string data, XmlDocument doc) : base(doc)
+        protected internal XmlProcessingInstruction(string target, string? data, XmlDocument doc) : base(doc)
         {
-            _target = target;
-            _data = data;
-        }
+            Debug.Assert(target != null, "All callsites should have already checked for null");
```

Comment:
This assert isn't valid because this constructor can be called by a derived class.

Reply:
good point, it's minor since we ship release binaries but will fix

```suggestion
```